Part 1: Refactor SVGLengthValue to use CSS values (css/values) for storage

nullhook opened the pull request:
#### 92b145d0fc92136385fec2fb32316036218c1115
<pre>
Refactor SVGLengthValue to use CSS values (css/values) for storage
<a href="https://bugs.webkit.org/show_bug.cgi?id=297096">https://bugs.webkit.org/show_bug.cgi?id=297096</a>
<a href="https://rdar.apple.com/157809123">rdar://157809123</a>

Reviewed by Darin Adler.

Replace direct storage of float value and SVGLengthType with with CSS strong value types.
This enables reuse of CSS parsing infrastructure and prepares for future support
of modern CSS units in SVG.

LayoutTests/imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-svglength-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/intersection-observer/svg-container-element-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/intersection-observer/svg-image-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/intersection-observer/svg-stroke-change-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-cap-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-ic-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-px-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-rem-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-rlh-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-viewport-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-1-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-2-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-3-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-5-expected.txt:
LayoutTests/svg/dom/length-list-parser-expected.txt:
Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
Source/WebCore/svg/SVGLengthValue.cpp:
(WebCore::adjustValueForPercentageStorage):
(WebCore::createVariantForLengthType):
(WebCore::SVGLengthValue::SVGLengthValue):
(WebCore::SVGLengthValue::setValueAsString):
(WebCore::SVGLengthValue::lengthType const):
(WebCore::SVGLengthValue::isZero const):
(WebCore::SVGLengthValue::isRelative const):
(WebCore::SVGLengthValue::valueAsPercentage const):
(WebCore::SVGLengthValue::valueInSpecifiedUnits const):
(WebCore::SVGLengthValue::valueAsString const):
(WebCore::SVGLengthValue::valueAsAtomString const):
(WebCore::SVGLengthValue::valueForBindings const):
(WebCore::SVGLengthValue::setValueInSpecifiedUnits):
(WebCore::SVGLengthValue::setValue):
(WebCore::SVGLengthValue::convertToSpecifiedUnits):
(WebCore::lengthTypeToString): Deleted.
(WebCore::parseLengthType): Deleted.
(WebCore::SVGLengthValue::fromCSSPrimitiveValue): Deleted.
(WebCore::SVGLengthValue::toCSSPrimitiveValue const): Deleted.
Source/WebCore/svg/SVGLengthValue.h:
(WebCore::SVGLengthValue::lengthType const): Deleted.
(WebCore::SVGLengthValue::isZero const): Deleted.
(WebCore::SVGLengthValue::isRelative const): Deleted.
(WebCore::SVGLengthValue::valueAsPercentage const): Deleted.
(WebCore::SVGLengthValue::valueInSpecifiedUnits const): Deleted.
(WebCore::SVGLengthValue::setValueInSpecifiedUnits): Deleted.

Canonical link: <a href="https://commits.webkit.org/298731@main">https://commits.webkit.org/298731@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c3194889c5369f3fd2de0e4d4c6be8aab8421180

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116470 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36131 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26682 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122527 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/67033 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118359 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36825 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44719 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88448 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42916 "An unexpected error occured. Recent messages:Printed configuration") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119419 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29363 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104480 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68891 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28434 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22586 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66208 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98738 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22747 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125676 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43364 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32566 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97154 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43729 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100689 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96949 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24676 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42243 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20148 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/39318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43252 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48843 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42718 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46058 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44423 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/80bb843985bdfc466ce3f60ed7172728549e8621. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

nullhook commented:
todo:
- is non-zero unitless values supported?
- what about calc?

Ahmad-S792 commented:
We might need to something similar in future for MathML but one thing at a time. Thanks for tackling it.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9f3daadd5b3cc3d5f7005b0839bf6b50cf596312. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2a076912f68ad1cc9c0ccb054b243f29e8958d4b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c4df008fc28423f5977acc7f54fc74117bbfb49e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fbddd11ef6b28214f7c60e5ed18ffbbe99bb0953. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

weinig commented:
I'm generally trying to move things off of the CSSValue hierarchy, so, if possible, using one of the CSS strong value types would be preferred here. 

Could this use something like `Variant<CSS::Number<>, CSS::LengthPercentage<>>` instead?

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7f02a6f42234d7dfd046a9ad953ffdaf3748904c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/db745e8a2ffd1f16d2cb58f0fca28d06686e6124. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/63df9a921702dde14c9c1bb72c6e3a36b4303ef3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c6284396cae5f96d126ae08e5b8ba9fa8e925d92. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/02cfaf4dcca04bda429ec84692b45e03f94ac2e5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/453068e7ec2502fb47f44a462256909ecc4c68eb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/648bef0f31ccf047aab377c1c74498c7f98f0b95. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

nullhook commented on Source/WebCore/svg/SVGLengthValue.cpp:
```diff
@@ -260,45 +219,62 @@ SVGLengthValue SVGLengthValue::blend(const SVGLengthValue& from, const SVGLength
 
-    return CSSPrimitiveValue::create(valueInSpecifiedUnits(), lengthTypeToPrimitiveType(lengthType()));
+bool SVGLengthValue::isZero() const
+{
+    return WTF::switchOn(m_value,
+        [](const auto& number) {
+            return number.isKnownZero();
+        },
+        [](const CSS::LengthPercentage<>& length) {
+            return length.isKnownZero();
+        }
+    );
```
can probably make this less verbose with `const auto& value` and in many places. not sure what's preferred here? 

fwiw, auto deduced in lambdas like this is bad for type hints.

@weinig

nullhook commented:
This is ready for review

darinadler commented on Source/WebCore/svg/SVGLengthValue.cpp:
```diff
@@ -184,22 +121,44 @@ static inline CSSUnitType lengthTypeToPrimitiveType(SVGLengthType lengthType)
+
 SVGLengthValue::SVGLengthValue(SVGLengthMode lengthMode, const String& valueAsString)
-    : m_lengthMode(lengthMode)
+    : m_value(CSS::Number<>(0))
```
I’d think we’d want this to be done with a default value rather than a value specified in the constructor.

nullhook replied:
makes sense. however, `m_value` here is fundamentally derived from three different types: length, percentage, and number. i don't think there is a universal default.

at this point in the constructor, we don't yet know what `m_value` represents until `setValueAsString` is called, so i'm not sure what an appropriate default would be.

maybe we should introduce another state to represent "uninitialized" — perhaps a std::monostate or an std::optional wrapper?

darinadler replied:
I don’t understand how this is different for an initial value for `m_value` in the constructor vs. the initial value as a default.

In practice, I believe this is the value if parsing fails, and otherwise whatever value we choose has no effect. Except maybe `setValueAsString` never fails, and then if so the value we choose is not even observable. I don’t think we should add `std::monostate`, because if we do we’ll have to add code to handle it everywhere. I think it’s good to always initialize every data member, unless it’s an important optimization to leave it uninitialized.

darinadler commented on Source/WebCore/svg/SVGLengthValue.cpp:
```diff
@@ -260,45 +219,62 @@ SVGLengthValue SVGLengthValue::blend(const SVGLengthValue& from, const SVGLength
+        [](const CSS::LengthPercentage<>& length) {
+            if (auto raw = length.raw()) {
+                using Unit = CSS::LengthPercentageUnit;
+                return raw->unit == Unit::Percentage
+                    || raw->unit == Unit::Em
+                    || raw->unit == Unit::Ex
+                    || raw->unit == Unit::Ch
+                    || raw->unit == Unit::Lh
+                    || raw->unit == Unit::Rem
+                    || raw->unit == Unit::Rex
+                    || raw->unit == Unit::Rlh
+                    || raw->unit == Unit::Rch;
```
Is this list of units somehow specific to SVG or is it a general “units that are relative” list? If the latter, then there should be a helper function elsewhere that we use, not a list written out here. It also seems like we’d want to use a `switch`.

karlcow replied:
It's unique to SVG. This is defined here. 
https://svgwg.org/svg2-draft/single-page.html#types-InterfaceSVGLength

It's a bit sad. Because to extend the list to new modern values, it requires acrobatic moves in terms of code. 

The spec has prose about it. 
> The use of numeric length unit type constants is an anti-pattern and new constant values will not be introduced for any other units or length types supported by [SVGLength](https://svgwg.org/svg2-draft/single-page.html#types-InterfaceSVGLength). If other types of lengths are supported and used, the [SVGLength](https://svgwg.org/svg2-draft/single-page.html#types-InterfaceSVGLength) uses the [SVG_LENGTHTYPE_UNKNOWN](https://svgwg.org/svg2-draft/single-page.html#types-__svg__SVGLength__SVG_LENGTHTYPE_UNKNOWN) unit type. See below for details on how the other properties of an [SVGLength](https://svgwg.org/svg2-draft/single-page.html#types-InterfaceSVGLength) operate with these types of lengths.

darinadler commented on Source/WebCore/svg/SVGLengthValue.cpp:
```diff
@@ -309,39 +285,143 @@ float SVGLengthValue::value(const SVGLengthContext& context) const
 {
-    return makeString(m_valueInSpecifiedUnits, lengthTypeToString(m_lengthType));
+    return WTF::switchOn(m_value,
+        [](const CSS::Number<>& number) {
```
Seems like we should use `const auto&` here to avoid the copied and pasted code.

darinadler commented on Source/WebCore/svg/SVGLengthValue.cpp:
```diff
@@ -350,35 +430,59 @@ ExceptionOr<void> SVGLengthValue::setValueAsString(StringView string)
-    return readCharactersForParsing(string, [&](auto buffer) -> ExceptionOr<void> {
-        auto convertedNumber = parseNumber(buffer, SuffixSkippingPolicy::DontSkip);
-        if (!convertedNumber)
+    // FIXME: SVG2 allows trailing spaces?
```
Can you write this in a way that’s clearer? What is the question? How would we change the code based on the answer? Why is it better to have a `FIXME` instead of a test case?

darinadler reviewed (approved):
Looks OK. Even though we do have many tests here, I suspect we need even more to cover all the many code paths.

nullhook commented on Source/WebCore/svg/SVGLengthValue.cpp:
```diff
@@ -350,35 +430,59 @@ ExceptionOr<void> SVGLengthValue::setValueAsString(StringView string)
+        if (!tokenRange.atEnd())
+            return Exception { ExceptionCode::SyntaxError };
+
+        // FIXME: Add support for calc
+        // For now, reject it
```
yes, it's `LayoutTests/svg/css/parse-calc-length.html` — this will be supported in a follow-up PR

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/88981344a9677dd513d76e3606967222738fc59c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

karlcow commented on LayoutTests/svg/dom/length-list-parser-expected.txt:
```diff
@@ -9,106 +9,95 @@ Parsed as 8 length(s) [ 28, -8, 0, 47, 4, 0, 35, 684 ]:  28	-8 -0.00000000000000
 Parsed as 2 length(s) [ 2, 0.5 ]: 2	.5,658-7 e ,8477553
 Parsed as 1 length(s) [ 2183 ]:  2183 9-0.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000191,59-- 70478.,566964784	e1-53	847e0.6e34805-0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001-0.000000000000000 ...
-Parsed as 5 length(s) [ 8, 2, 589, 0, -4 ]: 8 2 589	-45e58 -4	-0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001e58-++689.-98-,42 94564.,967-0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000011+ ,9+6-	.7 ...
+Parsed as 5 length(s) [ 8, 2, 589, -449999999999999959383614724858809976905065636837345328103424, -4 ]: 8 2 589	-45e58 -4	-0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001e58-++689.-98-,42 94564.,967-0.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 ...
```
Would there be value into separating this PR, into smaller individual bits. And not fixing everything in one shot. 
with each of the bits, the corresponding WPT?

nullhook replied:
i think it's better if i add a FIXME and address this in a follow-up. for now, we should return all these values as 0 for exponent input, which won’t break the current behavior.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e3efe67ddc27beffe46b2fe8e22f48cfe3265457. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9069ff1edcf9de0d8155a0e2633882699b755881. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/90111bd34499df01e6098228957ce0c44d6010d0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c3194889c5369f3fd2de0e4d4c6be8aab8421180. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49097

nullhook committed:
Refactor SVGLengthValue to use CSS values (css/values) for storage
https://bugs.webkit.org/show_bug.cgi?id=297096
rdar://157809123

Reviewed by Darin Adler.

Replace direct storage of float value and SVGLengthType with with CSS strong value types.
This enables reuse of CSS parsing infrastructure and prepares for future support
of modern CSS units in SVG.

LayoutTests/imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-svglength-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/intersection-observer/svg-container-element-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/intersection-observer/svg-image-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/intersection-observer/svg-stroke-change-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-cap-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-ic-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-px-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-rem-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-rlh-expected.txt:
LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGLength-viewport-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-1-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-2-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-3-expected.txt:
LayoutTests/svg/custom/svg-parse-overflow-5-expected.txt:
LayoutTests/svg/dom/length-list-parser-expected.txt:
Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
Source/WebCore/svg/SVGLengthValue.cpp:
(WebCore::adjustValueForPercentageStorage):
(WebCore::createVariantForLengthType):
(WebCore::SVGLengthValue::SVGLengthValue):
(WebCore::SVGLengthValue::setValueAsString):
(WebCore::SVGLengthValue::lengthType const):
(WebCore::SVGLengthValue::isZero const):
(WebCore::SVGLengthValue::isRelative const):
(WebCore::SVGLengthValue::valueAsPercentage const):
(WebCore::SVGLengthValue::valueInSpecifiedUnits const):
(WebCore::SVGLengthValue::valueAsString const):
(WebCore::SVGLengthValue::valueAsAtomString const):
(WebCore::SVGLengthValue::valueForBindings const):
(WebCore::SVGLengthValue::setValueInSpecifiedUnits):
(WebCore::SVGLengthValue::setValue):
(WebCore::SVGLengthValue::convertToSpecifiedUnits):
(WebCore::lengthTypeToString): Deleted.
(WebCore::parseLengthType): Deleted.
(WebCore::SVGLengthValue::fromCSSPrimitiveValue): Deleted.
(WebCore::SVGLengthValue::toCSSPrimitiveValue const): Deleted.
Source/WebCore/svg/SVGLengthValue.h:
(WebCore::SVGLengthValue::lengthType const): Deleted.
(WebCore::SVGLengthValue::isZero const): Deleted.
(WebCore::SVGLengthValue::isRelative const): Deleted.
(WebCore::SVGLengthValue::valueAsPercentage const): Deleted.
(WebCore::SVGLengthValue::valueInSpecifiedUnits const): Deleted.
(WebCore::SVGLengthValue::setValueInSpecifiedUnits): Deleted.

Canonical link: https://commits.webkit.org/298731@main

webkit-commit-queue commented:
Committed 298731@main (92b145d0fc92): <https://commits.webkit.org/298731@main>

Reviewed commits have been landed. Closing PR #49097 and removing active labels.
